### PR TITLE
[css-mediaqueries] Add 'prefers-reduced-data' (addresses #2370)

### PR DESCRIPTION
**Pull request by huijing**
Content pulled from discussion within #2370

### CODE REVIEW

**Comment by frivoal** [comment]
Thanks @huijing, this looks good. One suggestion: could you add a note saying that User Agents may consider setting this based on the same user or system preference as they use to set the Save-Data HTTP request header?

File: mediaqueries-5/Overview.bs [hunk 1]
```diff
@@ -688,6 +688,56 @@ Detecting a forced color palette: the 'forced-colors' feature</h3>
...
+Detecting the desire for reduced data usage when loading a page: the 'prefers-reduced-data' feature</h3>
+
+	<pre class='descdef mq'>
+	Name: prefers-reduced-data
+	Value: no-preference | reduce
+	For: @media
+	Type: discrete
+	</pre>
+
+	The 'prefers-reduced-data' media feature is used to detect is an user
```
Comment (yoavweiss) [inline mediaqueries-5/Overview.bs]: s/is an user/if a user/

File: mediaqueries-5/Overview.bs [hunk 2]
```diff
@@ -688,6 +688,56 @@ Detecting a forced color palette: the 'forced-colors' feature</h3>
...
+
+		<pre>
+		.image {
+			background-image: url("image/heavy.jpg");
+		}
+
+		@media (prefers-reduced-data: reduce) { 
+			/* Save-Data: On */
+			.image {
+				background-image: url("image/light.webp");
```
Comment (yoavweiss) [inline mediaqueries-5/Overview.bs]: There can be a future where we'd have user agents that support this MQ but not webp, so I would caution against coupling "lightweight" with browser support for formats. Having a way to do client side format negotiation for BG images would be a separate (and potentially worthwhile) use case IMO.

File: mediaqueries-5/Overview.bs (hunk 1 above)
Comment (huijing) [reply mediaqueries-5/Overview.bs]: Ah, typo, my bad.

File: mediaqueries-5/Overview.bs (hunk 2 above)
Comment (huijing) [reply mediaqueries-5/Overview.bs]: Should I switch out this use-case with another one from the thread? Or change the term "lightweight"?

Comment (yoavweiss) [reply mediaqueries-5/Overview.bs]: I'd just switch out the "webp" extension in favour of a "jpg"

Comment (huijing) [reply mediaqueries-5/Overview.bs]: Makes sense. Let me update that.

File: mediaqueries-5/Overview.bs [hunk 3]
```diff
@@ -688,6 +688,60 @@ Detecting a forced color palette: the 'forced-colors' feature</h3>
...
+
+		<pre>
+		.image {
+			background-image: url("images/light.jpg");
+		}
+
+		@media (prefers-reduced-data: reduce) { 
+			/* Save-Data: On */
+			.image {
+				background-image: url("images/heavy.jpg");
```
Comment (yoavweiss) [inline mediaqueries-5/Overview.bs]: I think "light" and "heavy" switched places :)

File: mediaqueries-5/Overview.bs [hunk 4]
```diff
@@ -688,6 +688,60 @@ Detecting a forced color palette: the 'forced-colors' feature</h3>
...
+			Indicates that user has expressed the preference for
+			lightweight alternate content.
+	</dl>
+	
+	The method by which the user expresses their preference can vary.
+	It might be a system-wide setting exposed by the Operating System,
+	or a setting controlled by the User Agent.
+	
+	Note: User Agents may consider setting this based on the same user 
+	or system preference as they use to set the Save-Data HTTP request
```
Comment (yoavweiss) [inline mediaqueries-5/Overview.bs]: You can point to https://wicg.github.io/savedata/ as a reference link

File: mediaqueries-5/Overview.bs (hunk 3 above)
Comment (huijing) [reply mediaqueries-5/Overview.bs]: I am sorry.

File: mediaqueries-5/Overview.bs (hunk 4 above)
Comment (huijing) [reply mediaqueries-5/Overview.bs]: Added link.

File: mediaqueries-5/Overview.bs (hunk 3 above)
Comment (yoavweiss) [reply mediaqueries-5/Overview.bs]: no worries :)

**Comment by frivoal** [comment]
Merged! thanks @huijing

**Comment by yoavweiss** [comment]
Yay!! 🎉

**Comment by ryantownsend** [comment]
Super excited about this! 😁

**Comment by huijing** [comment]
omg, this is the highlight of my day :)

**Pull request merged**